lib/paraactions/paraactions.ts

```diff
@@ -0,0 +1,790 @@
+        value: value.value,
+      });
+    }
+  }
+
+  return out;
+}
+
+/**
+ * Split arguments on commas, respecting parentheses/quotes.
+ */
+function splitArgs(input: string): string[] {
+  const parts: string[] = [];
+  let current = '';
+  let depth = 0;
+  let inSingle = false;
+  let inDouble = false;
+
+  for (let i = 0; i < input.length; i++) {
+    const ch = input[i];
+
+    if (ch === '\'' && !inDouble) {
+      inSingle = !inSingle;
+      current += ch;
+      continue;
+    }
+    if (ch === '"' && !inSingle) {
+      inDouble = !inDouble;
+      current += ch;
+      continue;
+    }
+
+    if (!inSingle && !inDouble) {
+      if (ch === '(') {
+        depth++;
+        current += ch;
+        continue;
+      }
+      if (ch === ')') {
+        depth = Math.max(0, depth - 1);
+        current += ch;
+        continue;
+      }
+      if (ch === ',' && depth === 0) {
+        parts.push(current);
+        current = '';
+        continue;
+      }
+    }
+
+    current += ch;
+  }
+
+  if (current) {
+    parts.push(current);
+  }
+
+  return parts;
+}
+
+/**
+ * Parse a scalar value:
+ */
+function parseValue(input: string): ParaValue {
+  const trimmed = input.trim();
+
+  // quoted string
+  if (
+    (trimmed.startsWith('\'') && trimmed.endsWith('\'')) ||
+    (trimmed.startsWith('"') && trimmed.endsWith('"'))
+  ) {
+    const inner = trimmed.slice(1, -1);
+    return {
+      valueKind: 'string',
+      value: inner,
+    };
+  }
+
+  if (trimmed === 'true') {
+    return { valueKind: 'boolean', value: true };
+  }
+  if (trimmed === 'false') {
+    return { valueKind: 'boolean', value: false };
+  }
+  if (trimmed === 'null') {
+    return { valueKind: 'null', value: null };
+  }
+
+  const num = Number(trimmed);
+  if (!Number.isNaN(num)) {
+    return { valueKind: 'number', value: num };
+  }
+
+  // identifier
+  return {
+    valueKind: 'identifier',
+    value: trimmed,
+  };
+}
+
+// ─────────────────────────────────────────────────────────────────────────────
+// When condition parsing
+// ─────────────────────────────────────────────────────────────────────────────
+
+/**
+ * Parse the contents of when(...), e.g.:
+ *   "direction='down'"
+ *   "direction='down', progressMin=0.3, progressMax=0.7"
+ *   "'down'"   // shorthand for direction='down'
+ */
+function parseWhenCondition(conditionText: string): GuardCondition {
+  const args = parseArgs(conditionText);
+  const named = args.filter((a): a is ParaNamedArg => a.kind === 'named');
+  const positional = args.filter(a => a.kind === 'positional');
+
+  const cond: GuardCondition = {};
+
+  // Named direction
+  const dirNamed = named.find(a => a.name === 'direction');
+  // Positional shorthand: when('down')
+  const firstPos = positional[0];
+
+  const rawDirection = dirNamed?.value ?? firstPos?.value;
+  if (rawDirection != null) {
+    const dirStr = String(rawDirection);
+    if (dirStr === 'up' || dirStr === 'down') {
+      cond.direction = dirStr;
+    }
```

Comment:
There should be an error or warning if direction is something other than 'up' or 'down'